client.go

```diff
@@ -598,45 +609,32 @@ func (c *Client[TTx]) Start(ctx context.Context) error {
 	// Monitor should be the first subprocess to start, and the last to stop.
 	// It's not part of the waitgroup because we need to wait for everything else
 	// to shut down prior to closing the monitor.
-	go c.monitor.Run()
+	//
+	// Unlike other services, it's given a background context so that it doesn't
+	// cancel on normal stops.
+	if err := c.monitor.Start(context.Background()); err != nil { //nolint:contextcheck
+		return err
+	}
 
 	// Receives job complete notifications from the completer and distributes
 	// them to any subscriptions.
 	c.completer.Subscribe(c.distributeJobCompleterCallback)
 
-	c.wg.Add(1)
-	go func() {
-		c.logStatsLoop(fetchNewWorkCtx)
-		c.wg.Done()
-	}()
-
-	if c.elector != nil {
-		go func() {
-			sub := c.elector.Listen()
-			defer sub.Unlisten()
-
-			for {
-				select {
-				case <-c.stopComplete: // stop complete
-					return
-
-				case <-workCtx.Done(): // stop started
-					return
-
-				case notification := <-sub.C():
-					c.handleLeadershipChange(ctx, notification)
-				}
-			}
-		}()
+	for _, service := range c.services {
+		if err := service.Start(fetchNewWorkCtx); err != nil {
+			// In case of error, stop any services that might have started. This
+			// is safe because even services that were never started will still
+			// tolerate being stopped.
+			startstop.StopAllParallel(c.services)
```

Comment:
We now try and stop anything that started even in the event of a start error, which is kind of nice.